### PR TITLE
Api integration for fetch groups

### PR DESCRIPTION
**Pull request by sanju-developer**

### CODE REVIEW

File: lib/screens/groups/bloc/groups_state.dart [hunk 1]
```diff
@@ -7,10 +7,27 @@ abstract class GroupsState extends Equatable {
...
+  GetGroupsSuccess(this.groups);
+
+  @override
+  List<Object> get props => [groups];
+}
+
+class GetGroupsFailure extends GroupsState {}
+
+// State for single group details
+class GetSignleGroupsDetailsInitial extends GroupsState {}
```
Comment (shubhamtanwar23) [inline lib/screens/groups/bloc/groups_state.dart]: typo `Single` in all the states

File: lib/screens/groups/widgets/group_page_body.dart [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  @override
+  void dispose() {
+    _groupsBloc.close();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => GroupsBloc(groupsRepository: groupsRepository),
```
Comment (shubhamtanwar23) [inline lib/screens/groups/widgets/group_page_body.dart]: you should pass the bloc created in `initState`

File: lib/screens/groups/widgets/group_page_body.dart [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+  @override
+  void initState() {
+    super.initState();
+    _groupsBloc = GroupsBloc(groupsRepository: GroupsRepository());
+    _groupsBloc.add(FetchGroups());
+  }
+
+  @override
+  void dispose() {
```
Comment (shubhamtanwar23) [inline lib/screens/groups/widgets/group_page_body.dart]: `BlocProvider` will close the bloc so we don't need to manually call it

**Review by shubhamtanwar23 — CHANGES_REQUESTED** [review]

File: lib/screens/groups/bloc/groups_state.dart (hunk 1 above)
Comment (sanju-developer) [reply lib/screens/groups/bloc/groups_state.dart]: Updated

File: lib/screens/groups/widgets/group_page_body.dart (hunk 3 above)
Comment (sanju-developer) [reply lib/screens/groups/widgets/group_page_body.dart]: Oh, I forgot

File: lib/screens/groups/widgets/group_page_body.dart (hunk 2 above)
Comment (sanju-developer) [reply lib/screens/groups/widgets/group_page_body.dart]: did not understand

**Comment by sanju-developer** [comment]
@shubhamtanwar23 , PR points updated. Please check

**Pull request merged**